hazelcast/src/main/java/com/hazelcast/internal/partition/operation/FinalizeMigrationOperation.java

```diff
@@ -39,6 +39,7 @@
  * Invoked locally on the source or destination of the migration to finalize the migration.
  * This will notify the {@link MigrationAwareService}s that the migration finished, updates the replica versions,
  * clears the migration flag and notifies the node engine when successful.
+ * There might be ongoing concurrent finalization operations for different or even for the same partition.
```

Comment:
`FinalizeMigrationOperation`s should not run concurrently for the same partition. Because it's executed on partition thread itself and a node cannot own multiple replicas of the same partition.

Reply:
Right, missed that. I'll fix the comment.